### PR TITLE
CI: Fixed PHP-CS-Fixer and Special-Char-Checker, Improved speed of Tests

### PR DESCRIPTION
**Pull request by catlauraherzog**
I once again improved our CI process. With this PR we are implementing and fixed several issues regarding our PHP-CS-Fixer and the Special-Char-Checker which both were broken in the first update. These changes were made:

1. The PHP-CS-Fixer and the Special-Char-Checker are only checking the PHP files which are changed within a pull request
2. The PHP-CS-Fixer now bailes if something is wrong and the checks fail with a certain message.
3. The Special-Char-Checker now works. Basically in the same way

This reduced the amount of time for a check by a lot. If this improvement is accepted for the branch release_6 I'll open a PR for release_7 and the trunk.

### CODE REVIEW

**Comment by catlauraherzog** [comment]
Could someone assist me with this error? https://github.com/ILIAS-eLearning/ILIAS/pull/3279/checks?check_run_id=2244011075#step:5:399

That's definitly not a problem of my changes. Maybe some merge dragged the failed Unit Test here.

File: .github/workflows/checks.yml [hunk 1]
```diff
@@ -3,10 +3,12 @@ on: [pull_request]
 jobs:
   tests:
     runs-on: ubuntu-latest
+    outputs:
+      all: ${{ steps.changes.outputs.all }}
     strategy:
       fail-fast: false
       matrix:
-        php: [7.2, 7.3, 7.4]
+        php: [7.3, 7.4]
```
Comment (mjansenDatabay) [inline .github/workflows/checks.yml]: @lauraquellmalz ILIAS 6 still supports PHP 7.2, so I am not sure if we should remove this here.

**Comment by mjansenDatabay** [comment]
> Could someone assist me with this error? https://github.com/ILIAS-eLearning/ILIAS/pull/3279/checks?check_run_id=2244011075#step:5:399
> 
> That's definitly not a problem of my changes. Maybe some merge dragged the failed Unit Test here.

@mbecker-databay I guess your latest changes regarding `assFormularQuestion` broke the unit tests.

**Comment by mjansenDatabay** [comment]
@lauraquellmalz 

> Could someone assist me with this error?

Done

**Comment by catlauraherzog** [comment]
@mjansenDatabay 
> Done

Thanks a lot!

File: .github/workflows/checks.yml (hunk 1 above)
Comment (catlauraherzog) [reply .github/workflows/checks.yml]: Reverted.

**Review by klees — COMMENTED** [review]
Hi @lauraquellmalz,

LGTM in general, I have one suggestion and one question. The suggestion:

* [ ] The suggestion: Lets deduplicate the statements to get the actually changed filed. E.g.: Put them in a separate script `get_changed_files_from_github.sh` and store them in the env. Maybe contact me to discuss options.
* [ ] The question: Why did you drop the check if the cs-fixer is actually installed?

Best regards!

**Comment by catlauraherzog** [comment]
Hi @klees 

> * [x]  The suggestion: Lets deduplicate the statements to get the actually changed filed. E.g.: Put them in a separate script `get_changed_files_from_github.sh` and store them in the env. Maybe contact me to discuss options.

Done, but slightly different. I used the common used file "functions.sh" which is used by the whole CI section.

> * [x]  The question: Why did you drop the check if the cs-fixer is actually installed?

Because the installation of the CS-Fixer is implemented via a github action. If this bailes the whole test suite will return an error nevertheless. The check is not necessary in my opinion. But it's not an issue for me to implement the check again.

Best, Laura

**Pull request merged**